csrc/python_frontend/python_bindings.cpp

```diff
@@ -171,6 +184,89 @@ std::vector<std::optional<bool>> computeContiguity(
   return contiguity;
 }
 
+// [ Note stride order and contiguity vector ]
+//
+// `stride order` vector corresponds to the order for each logical domain in
+//     physical memory;
+// `contiguity` vector to whether or not indexing could be collaped
+//     corresponding to each physical domain;
+//
+// e.g. Given size and stride as follow:
+//   sizes   = [2, 1, 3, 1, 4, 3]
+//   strides = [12, 4, 4, 4, 1, 0]
+// we would compute stride order as: [5, 4, 3, 2, 1, 0]. Since the original
+// stride is in descending order. Note that there's more than one way to define
+// a stride order when we have equal strides. In the context of index
+// collapsing, how we resolve that shouldn't matter, hence we just go with
+// preserving their original order. Similarly, we compute contiguity as: [True,
+// None, True, None, True, None], Since the physical order is the same as the
+// logical order, this one is trivial to compute.
+//
+// e.g. Given size and stride as follow:
+//   sizes   = [2, 3, 1, 5, 4]
+//   strides = [28, 4, 14, 0, 1]
+// stride_order would be: [4, 2, 3, 0, 1], marking the order of strides in the
+// vector. Meanwhile, contiguity would be computed on the physical domain, i.e.
+// on sorted sizes & strides.
+//   sorted_size    = [2, 1, 3, 4, 5]
+//   sorted_strides = [28, 14, 4, 1, 0]
+//   contiguity would be: [False, None, True, True, None]
+//
+// This function returns a tuple of <contiguity, stride_order>
+std::tuple<std::vector<std::optional<bool>>, std::vector<int64_t>>
+computeTensorDescriptor(
+    const std::vector<int64_t>& sizes,
```

Comment:
Computing stride order only requires the strides, and contiguity requires all three things: sizes, strides, and stride order. Should we split this into two functions with more descriptive names `computeStrideOrder` and `computeContiguity`?

Reply:
Good catch. We actually still have a `computeContiguity` api that computes contiguity following the logical domain order, rather than trying to compensate the stride order thing. I'm leaving that as-is for now and plan to migrate integration to use the new API.

On the topic of splitting the API. I intend to not do that and keep them as a whole. Hence the decision behind the naming here as `computeTensorDescriptor`.
I'm trying to avoid exposing all the nuts and bolts to user.Hopefully they would treat all the returned value here as an opaque thing; On top of that, `contiguity` vector by itself doesn't make much sense without accompanying the order where it was computed.